lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_l10n_helper.rb

```diff
@@ -116,6 +117,30 @@ def self.generate_strings_file_from_hash(translations:, output_path:)
           end
           File.write(output_path, builder.to_xml)
         end
+
+        # Downloads the export from GlotPress for a given locale and given filters.
+        #
+        # Also post-process the downloaded content to comment out any potential line with an empty (`""`) translation,
+        # to ensure those would fall back to the English/Base locale at runtime instead. These case should theoretically
+        # never happen as GlotPress should not include empty translations in its exports, but we've seen issues in the
+        # past, so better safe than sorry.
+        #
+        # @param [String] project_url The URL to the GlotPress project to export from, e.g. `"https://translate.wordpress.org/projects/apps/ios/dev"`
+        # @param [String] locale The GlotPress locale code to download strings for.
+        # @param [Hash{Symbol=>String}] filters The hash of filters to apply when exporting from GlotPress.
+        #                               Typical examples include `{ status: 'current' }` (the default) or `{ status: 'review' }`.
+        # @param [String, IO] destination The path or `IO`-like instance, where to write the downloaded file on disk.
```

Comment:
I'm not sure if it's common (as in "idiomatic"), but I've seen it in a couple of places.

It allows you to provide an IO if you already have one open instead of having just a path, not only for tests but also for some other contexts at call site where you might already have an IO opened and around to pass (e.g. if you want to concatenate various texts, or add a comment header at start of the file, you can `File.open`, then `f.write(header)`, then pass the `f` IO to the method to add the rest, then `f.close`. Granted, this is not super common nor really a use case for us here, but it's nice to have the flexibility, for tests or otherwise.

I would probably not add such a flexibility if it required a lot of additional code in the implementation just to support both. But given that here it doesn't cost anything (as `IO.copy_stream` already supports both), I figured it was worth it (especially given that it made the tests way nicer to write 😛 )

Reply:
👍 👍 👍